wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Poll/Editor.ts

```diff
@@ -0,0 +1,404 @@
+    }
+
+    const pollOptionInput = document.createElement('div') as HTMLDivElement;
+    pollOptionInput.classList.add('pollOptionInput');
+    listItem.appendChild(pollOptionInput);
+
+    const sortHandle = document.createElement('span') as HTMLSpanElement;
+    sortHandle.classList.add('icon', 'icon16', 'fa-arrows', 'sortableNodeHandle');
+    pollOptionInput.appendChild(sortHandle);
+
+    // buttons
+    const addButton = document.createElement('a') as HTMLAnchorElement;
+    listItem.setAttribute('role', 'button');
+    listItem.setAttribute('href', '#');
+    addButton.classList.add('icon', 'icon16', 'fa-plus', 'jsTooltip', 'jsAddOption', 'pointer');
+    addButton.setAttribute('title', Language.get('wcf.poll.button.addOption'));
+    addButton.addEventListener('click', () => this.createOption());
+    pollOptionInput.appendChild(addButton);
+
+    const deleteButton = document.createElement('a') as HTMLAnchorElement;
+    deleteButton.setAttribute('role', 'button');
+    deleteButton.setAttribute('href', '#');
+    deleteButton.classList.add('icon', 'icon16', 'fa-times', 'jsTooltip', 'jsDeleteOption', 'pointer');
+    deleteButton.setAttribute('title', Language.get('wcf.poll.button.removeOption'));
+    deleteButton.addEventListener('click', (ev) => this.removeOption(ev));
+    pollOptionInput.appendChild(deleteButton);
+
+    // input field
+    const optionInput = document.createElement('input') as HTMLInputElement;
+    optionInput.type = 'text';
+    optionInput.value = optionValue;
+    optionInput.maxLength = 255;
+    optionInput.addEventListener('keydown', (ev) => this.optionInputKeyDown(ev));
+    optionInput.addEventListener('click', () => {
+      // work-around for some weird focus issue on iOS/Android
+      if (document.activeElement !== optionInput) {
+        optionInput.focus();
+      }
+    });
+    pollOptionInput.appendChild(optionInput);
+
+    if (insertAfter !== null) {
+      optionInput.focus();
+    }
+
+    this.optionCount++;
+    if (this.optionCount === this.options.maxOptions) {
+      this.optionList.querySelectorAll('.jsAddOption').forEach((icon: HTMLSpanElement) => {
+        icon.classList.remove('pointer');
+        icon.classList.add('disabled');
+      });
+    }
+  }
+
+  /**
+   * Populates the option list with the current options.
+   */
+  private createOptionList(pollOptions: PollOption[]): void {
+    pollOptions.forEach((option) => {
+      this.createOption(option.optionValue, option.optionID);
+    });
+    
+    if (this.optionCount < this.options.maxOptions) {
+      this.createOption();
+    }
+  }
+
+  /**
+   * Handles validation errors returned by Ajax request.
+   */
+  private handleError(data: AjaxResponse): void {
+    switch (data.returnValues.fieldName) {
+      case this.wysiwygId + 'Poll_endTime':
+      case this.wysiwygId + 'Poll_maxVotes':
+        const fieldName = data.returnValues.fieldName.replace(this.wysiwygId + 'Poll_', '');
```

Comment:
To make eslint happy:
```suggestion
      case this.wysiwygId + 'Poll_maxVotes': {
        const fieldName = data.returnValues.fieldName.replace(this.wysiwygId + 'Poll_', '');
```